velox/substrait/SubstraitToVeloxExpr.cpp

```diff
@@ -319,8 +319,21 @@ SubstraitVeloxExprConverter::toVeloxExpr(
     const RowTypePtr& inputType) {
   auto substraitType = substraitParser_.parseType(castExpr.type());
   auto type = toVeloxType(substraitType->type);
-  // TODO add flag in substrait after. now is set false.
-  bool nullOnFailure = false;
+  auto failureBehavior =  castExpr.failure_behavior();
+  bool nullOnFailure;
+  switch (failureBehavior) {
+    case ::substrait::Expression_Cast_FailureBehavior_FAILURE_BEHAVIOR_UNSPECIFIED:
+      nullOnFailure = false;
+      break;
+    case ::substrait::Expression_Cast_FailureBehavior_FAILURE_BEHAVIOR_RETURN_NULL:
+      nullOnFailure = true;
+      break;
+    case ::substrait::Expression_Cast_FailureBehavior_FAILURE_BEHAVIOR_THROW_EXCEPTION:
+      nullOnFailure = false;
+      break;
+    default:
+      VELOX_FAIL("The given failure behavior is NOT supported!");
```

Comment:
Can we put the enum name here in error message as well?

Reply:
Yes, it makes sense to provide more hint to user. Just fixed.